400 Error, Unexpected end of JSON

I am receiving an HTTP 400 error with the description "Unexpected end of JSON input". The output seems to indicate that the task has established successful communication with DSS, but that the response from the API call is poorly formatted. Please advise.

Starting: Retrieve secret 12345 from Delinea Secret Server
==============================================================================
Task         : Delinea SS
Description  : Retrieve secrets from Delinea Secret Server
Version      : 0.0.5
Author       : Delinea
Help         : https://github.com/DelineaXPM/SecretServer.AzureDevOpsTask
==============================================================================
server.getSecret: 12345
server.getAccessToken: POST https://XYZ.secretservercloud.com/oauth2/token
server.sendRequest: GET https://XYZ.secretservercloud.com/api/v1/secrets/12345
server.getResponseContent: failed code 400
Unexpected end of JSON input
##[error]Unexpected end of JSON input
Finishing: Retrieve secret 12345 from Delinea Secret Server

--- a/DSSV1/index.ts
+++ b/DSSV1/index.ts
@@ -28,7 +28,7 @@ async function run() {
          * Make sure the secret id has been specified.
          */
         if (!taskParameters.secretId) {
-            tl.setResult(tl.TaskResult.Failed, "Secret id not specified");
+            tl.setResult(tl.TaskResult.Failed, "Secret id not specified", true);
             return;
         }
 
@@ -36,8 +36,8 @@ async function run() {
          * Get the secret from the server.
          */
         let secret: Secret = await server.getSecret(taskParameters.secretId);
-        if (!secret) {
-            tl.setResult(tl.TaskResult.Failed, "Secret not found");
+        if (!secret.id) {
+            tl.setResult(tl.TaskResult.Failed, "Secret not found", true);
             return;
         }
 
@@ -75,11 +75,11 @@ async function run() {
             }
         });
 
-        tl.setResult(tl.TaskResult.Succeeded, "");
+        tl.setResult(tl.TaskResult.Succeeded, "", true);
     }
     catch (err) {
         console.log(err.message);
-        tl.setResult(tl.TaskResult.Failed, err.message);
+        tl.setResult(tl.TaskResult.Failed, err.message, true);
     }
 }
 

--- a/DSSV1/operations/Server.ts
+++ b/DSSV1/operations/Server.ts
@@ -94,12 +94,12 @@ export class Server {
     /**
      * Gets the http response body.
      * @param response the http response
-     * @returns the reponse body as a string if the status code indicates success, empty string otherwise
+     * @returns the response body as a string if the status code indicates success, empty string otherwise
     */
     private async getResponseContent(response: httpClient.HttpClientResponse): Promise<string> {
         if (!this.isSuccessStatusCode(response.message.statusCode)) {
-            console.log(`server.getResponseContent: failed code ${response.message.statusCode}`);
-            return "";
+            console.log(`server.getResponseContent: failed code ${response.message.statusCode} and response body ${await response.readBody()}`);
+            return "{}";
         }
 
         let body: string = await response.readBody();

--- a/DSSV1/package.json
+++ b/DSSV1/package.json
@@ -4,22 +4,22 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "SET TASK_TEST_TRACE=1 && tsc && mocha .\\tests\\_suite.js"
   },
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "azure-pipelines-task-lib": "4.0.0-preview",
+    "azure-pipelines-task-lib": "^4.3.1",
     "mocha": "10.2.0",
     "typed-rest-client": "1.8.9"
   },
   "devDependencies": {
-    "@types/mocha": "10.0.0",
-    "@types/node": "16.18.0",
+    "@types/mocha": "^10.0.1",
+    "@types/node": "^20.2.1",
     "@types/q": "1.5.5",
     "sync-request": "6.1.0"
   },
-  "overrides" : {
+  "overrides": {
     "mockery": "2.1.0"
   }
 }

--- a/README.md
+++ b/README.md
@@ -80,10 +80,33 @@ Create a *failure_config.json* in the **DSSV1\tests** directory:
     "variablePrefix": "DSS_"
 }
 ```
-From the task directory **DSSV1**, run the following:
+From the task directory **DSSV1**, run:
 ```
-DSSV1> mocha .\tests\_suite.js
+npm run test
 ```
+## Debugging Unit Tests in VSCode
+Add this configuration to launch.json
+```
+{
+    "args": [
+        "-u",
+        "bdd",
+        "--timeout",
+        "999999",
+        "--colors",
+        "${workspaceFolder}/DSSV1/tests"
+    ],
+    "internalConsoleOptions": "openOnSessionStart",
+    "name": "Mocha Tests",
+    "program": "${workspaceFolder}/DSSV1/node_modules/mocha/bin/_mocha",
+    "request": "launch",
+    "skipFiles": [
+        "<node_internals>/**"
+    ],
+    "type": "node",
+    "cwd": "${workspaceRoot}/DSSV1/",
+}
+````
 
 # Packaging the extension
 Package the extension into a .vsix file using the following command from the repository root:

